src/mt/header.rs

```diff
@@ -0,0 +1,408 @@
+    #[test]
+    fn decode_all_true() {
+        let ans = DispositionFlags {
+            flush_queue: true,
+            send_ring_alert: true,
+            update_location: true,
+            high_priority: true,
+            assign_mtmsn: true,
+        };
+        assert_eq!(ans, DispositionFlags::decode(59));
+    }
+
+    #[test]
+    fn encode_all_false() {
+        let flags = DispositionFlags {
+            flush_queue: false,
+            send_ring_alert: false,
+            update_location: false,
+            high_priority: false,
+            assign_mtmsn: false,
+        };
+        assert_eq!(flags.encode(), 0);
+    }
+
+    #[test]
+    fn encode_flush_queue() {
+        let flags = DispositionFlags {
+            flush_queue: true,
+            send_ring_alert: false,
+            update_location: false,
+            high_priority: false,
+            assign_mtmsn: false,
+        };
+        assert_eq!(flags.encode(), 1);
+    }
+
+    #[test]
+    fn encode_send_ring_alert() {
+        let flags = DispositionFlags {
+            flush_queue: false,
+            send_ring_alert: true,
+            update_location: false,
+            high_priority: false,
+            assign_mtmsn: false,
+        };
+        assert_eq!(flags.encode(), 2);
+    }
+
+    #[test]
+    fn encode_update_location() {
+        let flags = DispositionFlags {
+            flush_queue: false,
+            send_ring_alert: false,
+            update_location: true,
+            high_priority: false,
+            assign_mtmsn: false,
+        };
+        assert_eq!(flags.encode(), 8);
+    }
+
+    #[test]
+    fn encode_high_priority() {
+        let flags = DispositionFlags {
+            flush_queue: false,
+            send_ring_alert: false,
+            update_location: false,
+            high_priority: true,
+            assign_mtmsn: false,
+        };
+        assert_eq!(flags.encode(), 16);
+    }
+
+    #[test]
+    fn encode_assign_mtmsn() {
+        let flags = DispositionFlags {
+            flush_queue: false,
+            send_ring_alert: false,
+            update_location: false,
+            high_priority: false,
+            assign_mtmsn: true,
+        };
+        assert_eq!(flags.encode(), 32);
+    }
+
+    #[test]
+    fn encode_all_true() {
+        let flags = DispositionFlags {
+            flush_queue: true,
+            send_ring_alert: true,
+            update_location: true,
+            high_priority: true,
+            assign_mtmsn: true,
+        };
+        assert_eq!(flags.encode(), 59);
+    }
+
+    #[test]
+    fn roundtrip_decode_encode() {
+        let combinations = vec![
+            1, 2, 3, 8, 9, 10, 11, 16, 17, 18, 19, 24, 25, 26, 27, 32, 33, 34, 35, 40, 41, 42, 43,
+            48, 49, 50, 51, 56, 57, 58, 59,
+        ];
+        for i in combinations {
+            assert_eq!(i, DispositionFlags::decode(i).encode())
+        }
+    }
+}
+
+/// Mobile Terminated Header
+///
+/// IEI: 0x41 [1] (Table 5-1)
```

Comment:
I'm not sure what this table reference refers to.

Reply:
Cleaned.